### DISCUSSION TITLE
Potential critical flaw?

### DISCUSSION THREAD

**Discussion by Daniel-DDV** [FlowiseAI/Flowise#2456] [Q&A]
Hi all,

After spending some time with Flowise and really loving it, yesterday I noticed I could feed one of my bots a 67k character input message which it happily devoured and ended up costing me 33 cents for just that message alone. I did set Max Tokens in ChatOpenAI node and was expecting this to be the total permitted amount of input/output tokens, but not so much. 

It needs no explanation that this is a way for users to really abuse a bot and crank up costs for the developer/owner really fast.

Am I doing something wrong or is this a critical flaw in FW that needs to be addressed asap?

**Comment by HenryHengZJ** [comment]
Can now specify `maxChars` to the chatbot https://github.com/FlowiseAI/FlowiseChatEmbed?tab=readme-ov-file#configuration

**Reply by Daniel-DDV** [reply]
Awesome, great solution. Thanks @HenryHengZJ
